src/main.c

```diff
@@ -103,30 +113,66 @@ int load_conf_file(char *file_path)
     return 0;
 }
 
+static void run_test(int test_n) {
+    for (int i = 0; i < tests_exclude.size; i++) {
+        // If a match is found in the exclusion list, then we skip the test.
+        if (test_n == vector_get(&tests_exclude, i)) {
+            test_n = -1;
+            break;
+        }
+    }
+    // Skip the test if test_n is a negative number.
+    if (test_n >= 0) {
+        kernel_thunk_table[test_n]();
+    }
+}
+
 static void run_tests()
 {
     print("Random seed used is %u", seed);
     if (tests_to_run.size == 0) {
-        print("No Specific tests specified. Running all tests (Single Pass).");
+        print("No specific tests were requested. Running all tests (Single Pass).");
+        if (tests_exclude.size) {
+            int remainder_size = 0;
+            for (int i = 0; i < tests_exclude.size; i++) {
+                if (kernel_thunk_table_size > vector_get(&tests_exclude, i)) {
+                    remainder_size++;
+                }
+            }
+            print("%d test(s) will be exclude.", remainder_size);
+        }
         print("-------------------------------------------------------------");
-        int table_size = ARRAY_SIZE(kernel_thunk_table);
-        for (int k=0; k<table_size; k++) {
-            kernel_thunk_table[k]();
+        for (int k = 0; k < kernel_thunk_table_size; k++) {
+            run_test(k);
```

Comment:
Contrary to the code above, it appears as if no tests are actually excluded in this loop?